sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -0,0 +1,252 @@
+    expected_encoding = cat_encoded[X_test]
+    assert_allclose(expected_encoding, X_trans)
+
+    # unknown
+    X_trans = enc.transform([unknown_X])
+    assert_allclose(X_trans, [[y_mean]])
+
+
+@pytest.mark.parametrize('categories, unknown_X', [
+    (
+        np.array([0, 1, 2], dtype=int),
+        np.array([4], dtype=int)
+    ),
+    (
+        np.array(['cat', 'dog', 'snake'], dtype=object),
+        np.array(['cow'], dtype=object)
+    ),
+])
+def test_zero_variance_group(categories, unknown_X):
+    # The first group is constant results in using the mean for the group
+    # as the encoding
+    X_int = np.array([[0] * 20 + [1] * 30 + [2] * 40], dtype=int).T
+    X_input = categories[X_int]
+
+    # The first group is a constant (10) and have no variance
+    y = np.array([10] * 20 + [-4] * 15 + [9] * 15 + [-6] * 30 + [25] * 10)
+
+    enc = TargetRegressorEncoder().fit(X_input, y)
+    X_test = np.array([[0]], dtype=int).T
+    X_input = categories[X_test]
+    X_trans = enc.transform(X_input)
+    assert_allclose(X_trans, [[10]])
+
+    # unknown
+    X_trans = enc.transform([unknown_X])
+    assert_allclose(X_trans, [[y.mean()]])
+
+
+@pytest.mark.parametrize('categories, unknown_X', [
+    (
+        np.array([0, 1, 2], dtype=int),
+        np.array([4], dtype=int)
+    ),
+    (
+        np.array(['cat', 'dog', 'snake'], dtype=object),
+        np.array(['cow'], dtype=object)
+    ),
+])
+def test_zero_variance_target(categories, unknown_X):
+    # if the target has zero variance, then the mean of the target is used.
+    X_int = np.array([[0] * 20 + [1] * 30 + [2] * 40], dtype=int).T
+    X_input = categories[X_int]
+    n_samples = X_input.shape[0]
+
+    y = np.ones(n_samples)
+
+    enc = TargetRegressorEncoder()
+    X_trans = enc.fit_transform(X_input, y)
+    expected_trans = np.full((n_samples, 1), fill_value=y.mean(), dtype=float)
+    assert_allclose(X_trans, expected_trans)
+
+    # unknown
+    X_trans = enc.transform([unknown_X])
+    assert_allclose(X_trans, [[y.mean()]])
+
+
+@pytest.mark.parametrize("X, categories", [
+    (
+        np.array([[0] * 10 + [1] * 10 + [3]],
+                 dtype=int).T,  # 3 is unknown
+        [[0, 1, 2]]),
+    (
+        np.array([['cat'] * 3 + ['dog'] * 4 + ['snake']],
+                 dtype=object).T,  # snake is unknown
+        [['dog', 'cat', 'cow']]
+    ),
+])
+def test_custom_categories(X, categories):
+    # Test custom categoires with unknow categories
+    rng = np.random.RandomState(42)
+    y = rng.uniform(low=-10, high=20, size=X.shape[0])
+
+    enc = TargetRegressorEncoder(categories=categories)
+    X_trans = enc.fit_transform(X, y)
+
+    # The last element is unknown
+    assert_allclose(X_trans[-1], [y.mean()])
+
+    assert len(enc.cat_encodings_) == 1
+    # unknown category seen during fitting is mapped to the mean
+    assert enc.cat_encodings_[0][-1] == pytest.approx(y.mean())
+
+
+def test_multiple_categories_santiy():
```

Comment:
typo

Reply:
Though did you mean multiple **features** instead? There are tests for multiple categories already